test/integration/controller/core/queuedworkload_controller_test.go

```diff
@@ -0,0 +1,152 @@
+		queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+		badQueue = testing.MakeQueue("badqueue", ns.Name).ClusterQueue("badCL").Obj()
+		gomega.Expect(k8sClient.Create(ctx, badQueue)).To(gomega.Succeed())
+		nonCreatedQueue = testing.MakeQueue("nonCreatedqueue", ns.Name).Obj()
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+		gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should update status when workloads are created", func() {
+		cases := []struct {
+			workload  *kueue.QueuedWorkload
+			condition kueue.QueuedWorkloadCondition
+		}{
+			{
+				// nil Queue
+				workload: testing.MakeQueuedWorkload("one", ns.Name).Request(corev1.ResourceCPU, "1").Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("Queue %s for workload doesn't exist", ""),
+				},
+			},
+			{
+				// Queue has not been created
+				workload: testing.MakeQueuedWorkload("two", ns.Name).Request(corev1.ResourceCPU, "1").
+					Queue(nonCreatedQueue.Name).Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("Queue %s for workload doesn't exist", nonCreatedQueue.Name),
+				},
+			},
+			{
+				// ClusterQueue non existent
+				workload: testing.MakeQueuedWorkload("three", ns.Name).Request(corev1.ResourceCPU, "1").
+					Queue(badQueue.Name).Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: fmt.Sprintf("ClusterQueue %s doesn't exist", badQueue.Spec.ClusterQueue),
+				},
+			},
+			{
+				// Request more resources than Ceiling limit
+				workload: testing.MakeQueuedWorkload("four", ns.Name).Request(corev1.ResourceCPU, "11").
+					Queue(queue.Name).Obj(),
+				condition: kueue.QueuedWorkloadCondition{
+					Type:    kueue.QueuedWorkloadAdmitted,
+					Status:  corev1.ConditionFalse,
+					Message: "workload didn't fit",
+				},
+			},
+		}
+
+		ginkgo.By("Creating workloads")
+		for _, tc := range cases {
+			gomega.Expect(k8sClient.Create(ctx, tc.workload)).To(gomega.Succeed())
+		}
+
+		// skip last case since it won't have status before being admitted
+		for _, tc := range cases[0:2] {
+			gomega.Eventually(func() string {
+				var updatedQueueWorkload kueue.QueuedWorkload
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(tc.workload), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions[0].Message
+			}, framework.Timeout, framework.Interval).Should(testing.Equal(tc.condition.Message))
+		}
+
+		ginkgo.By("Finishing workloads")
```

Comment:
This doesn't need to be tested (you are directly adding the Finished condition, so there is nothing to test).